### DISCUSSION TITLE
Looking for a collaborative editor that can be integrated or embeded in a REMIX app

### DISCUSSION THREAD

**Discussion by timougue** [toeverything/blocksuite#235] [Ideas]
Hi, 

I am discovering the AFFiNE and BlockSuite projects, they are awesome projects!

I am planing the development of a large collaborative app with potentially a large number of modular addons (notion-line feature, chat feature, and more specific ones...) I am doing now tech screening and I am looking for a **collaborative editor** that can be integrated or embedded in a REMIX app.

Do you think AFFiNE and/or BlockSuite could do the job?

Thx and cheers

**Comment by doodlewind** [comment]
Hi, it's great to see you are interested in AFFiNE and BlockSuite!

Technically, BlockSuite *will* become a general-purpose collaborative editing *framework*. Still, for now, it *is* already a *collaborative editor*, with all the UI components built for AFFiNE out of the box. In other words, the `blocks` and `editor` packages in BlockSuite are built for AFFiNE (a full-featured editor) for now but are designed to be replaceable.

You can reuse our editor in Remix. The editor itself is nothing more than a standard Web component (check out DevTools in [AFFiNE Alpha](https://pathfinder.affine.pro/)!). All custom blocks can be defined and reused in our design as Web components. This makes the editor framework agnostic. In fact, in the AFFiNE repository, we are using React for building the site, installing BlockSuite editor as a standard dependency.

As for Notion-like features, we are already working on the image block, code block, and slash menu. These planned features will be shipped in December, and we will also support the [provider-based](https://github.com/toeverything/blocksuite/discussions/225) collaborative editing feature in our Web version altogether by then.

After the December version, our development resources will mainly focus on the whiteboard (edgeless mode). The target is that within two major versions (about four months) next year, we should be able to support similar common feature sets compared with Notion. And this editor will always be open-source and free to use out of the box.

For more information, welcome to join our [discord group](https://discord.gg/apQ7XRgh), and we can even arrange further meetings with you if necessary :D

**Reply by northvip** [reply]
That sounds great. Could you tell me when the December version will be released?

**Reply by doodlewind** [reply]
Our current plan is to have major new features (image block, code block, inline hover menu, collaboration, page tree...) completed separately at around **December 10**, an integrated alpha version (tested internally) on **December 20**, and a release to [online demo](https://pathfinder.affine.pro) around **Christmas** 🎄. It would be a significant enhancement. Stay tuned!

**Reply by northvip** [reply]
I wonder if the hover menu will include delimited comments in this month's release？

**Reply by doodlewind** [reply]
The comment feature requires non-trivial data structure design and comprehensive testing. So I'm afraid it wouldn't inside the scope of the December version. But it's really important for collaboration and we have kept that in mind, and will let you know for further updates.
